---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r setup, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "man/figures/README-",
  out.width = "100%"
)
```

# gwasrapidd <img src="man/figures/logo.svg" align="right" height=140/>

[![Travis build status](https://travis-ci.org/ramiromagno/gwasrapidd.svg?branch=master)](https://travis-ci.org/ramiromagno/gwasrapidd)
[![AppVeyor build status](https://ci.appveyor.com/api/projects/status/github/ramiromagno/gwasrapidd?branch=master&svg=true)](https://ci.appveyor.com/project/ramiromagno/gwasrapidd)
[![codecov](https://codecov.io/gh/ramiromagno/gwasrapidd/branch/master/graph/badge.svg)](https://codecov.io/gh/ramiromagno/gwasrapidd)
[![License: MIT](https://img.shields.io/badge/License-MIT-yellow.svg)](https://opensource.org/licenses/MIT)
[![lifecycle](https://img.shields.io/badge/lifecycle-experimental-orange.svg)](https://www.tidyverse.org/lifecycle/#experimental)
`r badger::badge_devel("ramiromagno/gwasrapidd", "blue")`

The goal of gwasrapidd is to retrieve data from the [NHGRI-EBI Catalog](https://www.ebi.ac.uk/gwas) of published genome-wide association studies via the [REST API](https://www.ebi.ac.uk/gwas/docs/api).

Get started by reading the [gwasrapidd documentation](https://rmagno.eu/gwasrapidd/articles/gwasrapidd.html).

## Installation

You can install the current, experimental version of gwasrapidd with:

``` r
# install.packages("devtools")
devtools::install_github("ramiromagno/gwasrapidd")
```

## Cheatsheet

<a href="https://github.com/ramiromagno/gwasrapidd/blob/master/inst/cheatsheet/gwasrapidd_cheatsheet.pdf"><img src="https://raw.githubusercontent.com/ramiromagno/gwasrapidd/master/inst/cheatsheet/gwasrapidd_cheatsheet_thumbs.png" width="420" height="300"/></a>  


## Example

Get the study related to triple-negative breast cancer:

```{r example1}
library(gwasrapidd)
studies <- get_studies(efo_trait = 'triple-negative breast cancer')
studies@studies[1:4]
```

Find associated variants:

```{r example2}
variants <- get_variants(study_id = 'GCST002305')
variants@variants[c('variant_id', 'functional_class')]
```

## Contributors

Please note that the 'gwasrapidd' project is released with a [Contributor Code of Conduct](CODE_OF_CONDUCT.md). By contributing to this project, you agree to abide by its terms.
