---
title: "Frequently asked questions"
author: ""
date: ""
output:
  rmarkdown::html_vignette:
vignette: >
  %\VignetteIndexEntry{Frequently asked questions}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

## 1 | How to be sure that I can establish a connection to the GWAS Catalog server?

You can check that gwasrapidd is able to connect to [https://www.ebi.ac.uk](https://www.ebi.ac.uk) my making a connection attempt with the function `is_ebi_reachable()`:

```{r eval=FALSE}
library(gwasrapidd)
is_ebi_reachable()
```

Returns `TRUE` if the connection is possible, or `FALSE` otherwise. If the connection is not possible, use the parameter `chatty = TRUE` to learn at what point the connection is failing.

```{r eval=FALSE}
library(gwasrapidd)
is_ebi_reachable(chatty = TRUE)
```


## 2 | What resources is the GWAS Catalog database currently mapped against?

The GWAS Catalog is mapped against [Ensembl](https://www.ensembl.org), [dbSNP](https://www.ncbi.nlm.nih.gov/snp/) and a specific [assembly version](https://www.ncbi.nlm.nih.gov/grc/human) of the human genome. You can get this info with `get_metadata()`:

```{r}
library(gwasrapidd)
get_metadata()
```

## 3 | What is the difference between a trait and a reported trait?

There are two levels of trait description in the GWAS Catalog: (EFO) trait and reported trait.

Studies are assigned one or more terms from the [Experimental Factor Ontology (EFO)](https://www.ebi.ac.uk/efo/), i.e., an EFO trait, or simply trait, that best represents the phenotype under investigation.

In addition, each study is also assigned a free text reported trait. This is written by the GWAS Catalog curators and reflects the author language, and where necessary, it includes more specific and detailed description of the experimental design, e.g., interaction studies or studies with a background trait.

As an example take the study with accession identifier `GCST000206` by EM Behrens et al. (2008). We can get the EFO trait with `get_traits()` and the reported trait with `get_studies()`:

The (EFO) trait for the Behrens study is `chronic childhood arthritis`:

```{r}
library(gwasrapidd)
efo_trait <- get_traits(study_id = 'GCST000206')
efo_trait@traits$trait
```

whereas the reported trait is `Arthritis (juvenile idiopathic)`:

```{r}
study <- get_studies(study_id = 'GCST000206')
study@studies$reported_trait
```

## 4 | Genomic coordinates of genomic contexts seem to be wrong?

The REST API response for variants contains an element named *genomic contexts*. This element is mapped onto the table `genomic_contexts` of a [variants](https://rmagno.eu/gwasrapidd/reference/variants-class.html) S4 object.

Now, there is indeed a server-side bug with the column `chromosome_position` of the `genomic_contexts` table: the chromosome position returned is that of the variant and not of the gene (*genomic context*) as it should be.

The GWAS Catalog team is aware of this bug, and they plan to fix it, eventually. For the time being, just do not rely on `chromosome_position` of the `genomic_contexts` table.

## 5 | How to search for variants within a certain genomic region?

### Single genomic range

For this you may use the function `get_variants()` with parameter `genomic_range`.

For example, to search for variants in chromosome Y in the interval 14692000--14695000, you start by defining a list of 3 elements: `chromosome`, `start` and `end` that specify your genomic range:

```{r}
library(gwasrapidd)
# 'chromosome' names are case sensitive, and should be uppercase.
# 'start' and 'end' positions should be integer vectors.
my_genomic_range <- list(
  chromosome = 'Y',
  start = 14692000L,
  end = 14695000L)
```

Now you can use `my_genomic_range` to retrieve the variants:

```{r}
chr_Y_variants <- get_variants(genomic_range = my_genomic_range)
chr_Y_variants@variants[c('variant_id', 'functional_class')]
```

### Multiple genomic ranges

To search in multiple regions, construct your genomic range list with those locations just like in the previous example. For example, let's search now for variants in chromosome X and Y, both in range 13000000--15000000:

```{r}
my_genomic_range <- list(
  chromosome = c('X', 'Y'),
  start = c(13000000L, 13000000L),
  end = c(15000000L, 15000000L))

chr_XY_variants <- get_variants(genomic_range = my_genomic_range)
chr_XY_variants@variants[c('variant_id',
                           'chromosome_name',
                           'chromosome_position')]
```

### Searching variants by cytogenetic regions

To search for variants within a cytogenetic band you can use the parameter `cytogenetic_band` of `get_variants()`. Here is an example, again for chromosome Y, using the cytogenetic band `'Yq11.221'` as query:

```{r}
my_variants <- get_variants(cytogenetic_band = 'Yq11.221')
my_variants@variants[c('variant_id',
                       'chromosome_name',
                       'chromosome_position')]
```

How to know what are the cytogenetic bands for querying? We provide a dataset (dataframe) named `cytogenetic_bands` that you can use:

```{r}
# ?cytogenetic_bands for more details.
cytogenetic_bands
```

Let's say you want to search for all variants in the shorter arm (p) of chromosome 21, you can take advantage of the `cytogenetic_bands` to get all the corresponding cytogenetic band names:

```{r}
# Install package dplyr if you do not have it.
chr21_p_bands <- dplyr::filter(cytogenetic_bands, grepl('^21p', cytogenetic_band)) %>%
  dplyr::pull(cytogenetic_band)
chr21_p_bands
```

Now search by `cytogenetic_band`:
```{r}
my_variants <- get_variants(cytogenetic_band = chr21_p_bands)
my_variants@variants[c('variant_id',
                       'chromosome_name',
                       'chromosome_position',
                       'chromosome_region')]
```

## 6 | Genomic range for an entire chromosome?

You can get the total length of a chromosome by using the provided dataset: `cytogenetic_bands`. Here is an example for chromosome 15:

```{r}
library(gwasrapidd)
# Install dplyr first.
dplyr::filter(cytogenetic_bands, chromosome == '15') %>%
  dplyr::summarise(chromosome = dplyr::first(chromosome),
                   start = min(start),
                   end = max(end)
                   )
```

## 7 | How to keep track of which queries generated which results?

Currently, there is not an implemented solution in gwasrapidd. For example, if you search for variants by EFO identifier (`efo_id`):

```{r}
library(gwasrapidd)
my_efo_ids <- c('EFO_0005543', 'EFO_0004762')
my_variants <- get_variants(efo_id = my_efo_ids)
```

So it is not immediately obvious which variants resulted from the query `'EFO_0005543'` or `'EFO_0004762'`.

A possible workaround is to make multiple independent queries and save your results in a list whose names are the respective queries:

```{r}
# Install purrr first.
# Add names to my_efo_ids
names(my_efo_ids) <- my_efo_ids
my_variants <- purrr::map(my_efo_ids, ~ get_variants(efo_id = .x))
```

Now you can see which variants are associated with each EFO identifier.

For `'EFO_0005543'` we got the following variants:

```{r}
my_variants[['EFO_0005543']]@variants$variant_id
```

And for `'EFO_0004762'`:

```{r}
my_variants[['EFO_0004762']]@variants$variant_id
```

## 8 | How to combine results from multiple queries?

The four main retrieval functions `get_studies()`, `get_associations()`, `get_variants()` and `get_traits()` all allow you to search multiple criteria at once. You can then combine results in an OR or AND fashion using the parameter `set_operation`.

Use `set_operation = 'union'` to combine results in an OR fashion:

```{r}
my_variants_OR <- get_variants(
  efo_trait = 'triple-negative breast cancer',
  gene_name = 'MDM4',
  set_operation = 'union')

my_variants_OR@variants[c('variant_id',
                       'chromosome_name',
                       'chromosome_position',
                       'chromosome_region')]
```

Or use `set_operation = 'intersection'` to combine results in an AND fashion:

```{r}
my_variants_AND <- get_variants(
  efo_trait = 'triple-negative breast cancer',
  gene_name = 'MDM4',
  set_operation = 'intersection')

my_variants_AND@variants[c('variant_id',
                       'chromosome_name',
                       'chromosome_position',
                       'chromosome_region')]
```



<!--
## 9 | How to exclude results for specific queries?

The approach here is to search first using all your queries, including those queries for results you want to exclude, and then remove them with `setdiff`.

Let's get all variants in the cytogenetic region

## 5 | How to search by EFO identifier, including its child terms?

The [Experimental Factor Ontology (EFO)](https://www.ebi.ac.uk/efo/) identifiers can have one or more child terms as part of its EFO hierarchy. These child terms have a narrower meaning, and oftentimes one would like to search by an EFO identifier and also by all its child identifiers. This functionality is provided by the [GWAS Catalog Web User Interface](https://www.ebi.ac.uk/gwas/) in a check box, see image below.

```{r echo=FALSE, fig.align="left"}
# All defaults
knitr::include_graphics("../man/figures/child_trait.png")
```

However there is no such option for the REST API. To circumvent this problem you can use the function `get_child_efo()` that will get all child terms of a given EFO identifier. Here's an example with the EFO identifier `'EFO_0005543'` (glioma):


```{r}
library(gwasrapidd)
my_efo_id <- 'EFO_0005543'
my_studies <- get_studies(efo_id = my_efo_id)
n(my_studies)
```

```{r}
child_efo_ids <- get_child_efo(efo_id = my_efo_id)
my_studies_w_children <- get_studies(efo_id = c(my_efo_id, child_efo_ids[['my_efo_id']]))
n(my_studies_w_children)
```

We find `r n(my_studies)` studies using `'EFO_0005543'` only, and `r n(my_studies_w_children)` studies if we include also the child EFO identifiers.
--> 
