---
title: "Getting started with gwasrapidd"
author: ""
date: ""
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Getting started with gwasrapidd}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
bibliography: gwasrapidd.bib
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

## Getting started

The GWAS Catalog is a service provided by the EMBL-EBI and NHGRI that offers a
manually curated and freely available database of published genome-wide
association studies (GWAS). The Catalog website and infrastructure is hosted by
the [EMBL-EBI](https://www.ebi.ac.uk).

There are three ways to access the Catalog database:

- Via the Web interface at
[https://www.ebi.ac.uk/gwas](https://www.ebi.ac.uk/gwas)
- By downloading a
database dump in TSV or RDF/OWL formats:
[https://www.ebi.ac.uk/gwas/docs/file-downloads](https://www.ebi.ac.uk/gwas/docs/file-downloads)
- Or through the GWAS Catalog REST API:
[https://www.ebi.ac.uk/gwas/rest/docs/api](https://www.ebi.ac.uk/gwas/rest/docs/api)

gwasrapidd facilitates the access to the Catalog via the Catalog REST API,
allowing you to programmatically retrieve data directly into R.

## Usage

The Catalog REST API provides four types of data objects for retrieval:

- Studies
- Associations
- Variants
- Traits

gwasrapidd provides four corresponding functions to get each of the object
types:

- `get_studies()`
- `get_associations()`
- `get_variants()`
- `get_traits()`

## Examples

### Example 1 | Finding Risk Alleles Associated with Autoimmune Disease

As a first example, take the work by @Light2014. In this work the authors focused on variants that had been previously reported in genome-wide association studies (GWAS) for autoimmune disease. 

With gwasrapidd we can interrogate the GWAS Catalog for the study/studies by searching by *autoimmune disease* (an EFO trait). To do that let's load gwasrapidd first:

```{r}
library(gwasrapidd)
```

Then query the GWAS Catalog by EFO trait:

```{r cache=TRUE}
my_studies <- get_studies(efo_trait = 'autoimmune disease')
```

We can now check how many GWAS studies we got back:

```{r}
n(my_studies)
my_studies@studies$study_id
```

Apparently only one (`r my_studies@studies$study_id`). Let's see the associated publication title:

```{r}
my_studies@publications$title
```

Sounds interesting... To quickly browse to the PubMed entry for this publication:

```{r eval=FALSE}
# This launches your web browser at https://www.ncbi.nlm.nih.gov/pubmed/26301688
open_in_pubmed(my_studies@publications$pubmed_id) 
```

Now if we want to know the variants previously associated with autoimmune disease as used by @Light2014, we need to retrieve statistical association information on these variants and then filter them based on the same level of significance ($P < 1\times 10^{-6}$) [@Light2014].

So let's start by getting the associations by `study_id`:

```{r cache=TRUE}
# You could have also used get_associations(efo_trait = 'autoimmune disease')
my_associations <- get_associations(study_id = "GCST003097")
```

Seemingly there are `r n(my_associations)` associations.

```{r}
n(my_associations)
```

However not all variants meet the level of significance ($P < 1\times 10^{-6}$) as required by @Light2014:

```{r}
dplyr::filter(my_associations@associations, pvalue < 1e-6) %>% # Filter by p-value
  dplyr::pull(association_id) %>% # Extract column association_id
  as.character() -> association_ids # Convert to character
# Extract associations by association id
my_associations2 <- my_associations[association_ids]
n(my_associations2)
```

Of the `r n(my_associations)` associations found in GWAS Catalog, `r n(my_associations2)` meet the p-value threshold of $1\times 10^{-6}$. Here are the variants, risk allele and risk frequency:

```{r}
my_associations2@risk_alleles[c('variant_id', 'risk_allele', 'risk_frequency')] %>% print(n = Inf)
```


## References
