<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Frequently asked questions â€¢ gwasrapidd</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Frequently asked questions">
<meta property="og:description" content="">
<meta property="og:image" content="https://rmagno.eu/gwasrapidd/logo.png">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">gwasrapidd</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.7</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/gwasrapidd.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/faq.html">Frequently asked questions</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/ramiromagno/gwasrapidd">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Frequently asked questions</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/ramiromagno/gwasrapidd/blob/master/vignettes/faq.Rmd"><code>vignettes/faq.Rmd</code></a></small>
      <div class="hidden name"><code>faq.Rmd</code></div>

    </div>

    
    
<div id="how-to-be-sure-that-i-can-establish-a-connection-to-the-gwas-catalog-server" class="section level2">
<h2 class="hasAnchor">
<a href="#how-to-be-sure-that-i-can-establish-a-connection-to-the-gwas-catalog-server" class="anchor"></a>1 | How to be sure that I can establish a connection to the GWAS Catalog server?</h2>
<p>You can check that gwasrapidd is able to connect to <a href="https://www.ebi.ac.uk">https://www.ebi.ac.uk</a> my making a connection attempt with the function <code><a href="../reference/is_ebi_reachable.html">is_ebi_reachable()</a></code>:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" title="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(gwasrapidd)</a>
<a class="sourceLine" id="cb1-2" title="2"><span class="kw"><a href="../reference/is_ebi_reachable.html">is_ebi_reachable</a></span>()</a></code></pre></div>
<p>Returns <code>TRUE</code> if the connection is possible, or <code>FALSE</code> otherwise. If the connection is not possible, use the parameter <code>chatty = TRUE</code> to learn at what point the connection is failing.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" title="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(gwasrapidd)</a>
<a class="sourceLine" id="cb2-2" title="2"><span class="kw"><a href="../reference/is_ebi_reachable.html">is_ebi_reachable</a></span>(<span class="dt">chatty =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
</div>
<div id="what-resources-is-the-gwas-catalog-database-currently-mapped-against" class="section level2">
<h2 class="hasAnchor">
<a href="#what-resources-is-the-gwas-catalog-database-currently-mapped-against" class="anchor"></a>2 | What resources is the GWAS Catalog database currently mapped against?</h2>
<p>The GWAS Catalog is mapped against <a href="https://www.ensembl.org">Ensembl</a>, <a href="https://www.ncbi.nlm.nih.gov/snp/">dbSNP</a> and a specific <a href="https://www.ncbi.nlm.nih.gov/grc/human">assembly version</a> of the human genome. You can get this info with <code><a href="../reference/get_metadata.html">get_metadata()</a></code>:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" title="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(gwasrapidd)</a>
<a class="sourceLine" id="cb3-2" title="2"><span class="kw"><a href="../reference/get_metadata.html">get_metadata</a></span>()</a>
<a class="sourceLine" id="cb3-3" title="3"><span class="co">#&gt; $ensembl_release_number</span></a>
<a class="sourceLine" id="cb3-4" title="4"><span class="co">#&gt; [1] 96</span></a>
<a class="sourceLine" id="cb3-5" title="5"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb3-6" title="6"><span class="co">#&gt; $genome_build_version</span></a>
<a class="sourceLine" id="cb3-7" title="7"><span class="co">#&gt; [1] "GRCh38.p12"</span></a>
<a class="sourceLine" id="cb3-8" title="8"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb3-9" title="9"><span class="co">#&gt; $dbsnp_version</span></a>
<a class="sourceLine" id="cb3-10" title="10"><span class="co">#&gt; [1] 151</span></a>
<a class="sourceLine" id="cb3-11" title="11"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb3-12" title="12"><span class="co">#&gt; $usage_start_date</span></a>
<a class="sourceLine" id="cb3-13" title="13"><span class="co">#&gt; [1] "2019-04-08 19:00:02 UTC"</span></a></code></pre></div>
</div>
<div id="what-is-the-difference-between-a-trait-and-a-reported-trait" class="section level2">
<h2 class="hasAnchor">
<a href="#what-is-the-difference-between-a-trait-and-a-reported-trait" class="anchor"></a>3 | What is the difference between a trait and a reported trait?</h2>
<p>There are two levels of trait description in the GWAS Catalog: (EFO) trait and reported trait.</p>
<p>Studies are assigned one or more terms from the <a href="https://www.ebi.ac.uk/efo/">Experimental Factor Ontology (EFO)</a>, i.e., an EFO trait, or simply trait, that best represents the phenotype under investigation.</p>
<p>In addition, each study is also assigned a free text reported trait. This is written by the GWAS Catalog curators and reflects the author language, and where necessary, it includes more specific and detailed description of the experimental design, e.g., interaction studies or studies with a background trait.</p>
<p>As an example take the study with accession identifier <code>GCST000206</code> by EM Behrens et al.Â (2008). We can get the EFO trait with <code><a href="../reference/get_traits.html">get_traits()</a></code> and the reported trait with <code><a href="../reference/get_studies.html">get_studies()</a></code>:</p>
<p>The (EFO) trait for the Behrens study is <code>chronic childhood arthritis</code>:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" title="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(gwasrapidd)</a>
<a class="sourceLine" id="cb4-2" title="2">efo_trait &lt;-<span class="st"> </span><span class="kw"><a href="../reference/get_traits.html">get_traits</a></span>(<span class="dt">study_id =</span> <span class="st">'GCST000206'</span>)</a>
<a class="sourceLine" id="cb4-3" title="3">efo_trait<span class="op">@</span>traits<span class="op">$</span>trait</a>
<a class="sourceLine" id="cb4-4" title="4"><span class="co">#&gt; [1] "chronic childhood arthritis"</span></a></code></pre></div>
<p>whereas the reported trait is <code>Arthritis (juvenile idiopathic)</code>:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" title="1">study &lt;-<span class="st"> </span><span class="kw"><a href="../reference/get_studies.html">get_studies</a></span>(<span class="dt">study_id =</span> <span class="st">'GCST000206'</span>)</a>
<a class="sourceLine" id="cb5-2" title="2">study<span class="op">@</span>studies<span class="op">$</span>reported_trait</a>
<a class="sourceLine" id="cb5-3" title="3"><span class="co">#&gt; [1] "Arthritis (juvenile idiopathic)"</span></a></code></pre></div>
</div>
<div id="genomic-coordinates-of-genomic-contexts-seem-to-be-wrong" class="section level2">
<h2 class="hasAnchor">
<a href="#genomic-coordinates-of-genomic-contexts-seem-to-be-wrong" class="anchor"></a>4 | Genomic coordinates of genomic contexts seem to be wrong?</h2>
<p>The REST API response for variants contains an element named <em>genomic contexts</em>. This element is mapped onto the table <code>genomic_contexts</code> of a <a href="https://rmagno.eu/gwasrapidd/reference/variants-class.html">variants</a> S4 object.</p>
<p>Now, there is indeed a server-side bug with the column <code>chromosome_position</code> of the <code>genomic_contexts</code> table: the chromosome position returned is that of the variant and not of the gene (<em>genomic context</em>) as it should be.</p>
<p>The GWAS Catalog team is aware of this bug, and they plan to fix it, eventually. For the time being, just do not rely on <code>chromosome_position</code> of the <code>genomic_contexts</code> table.</p>
</div>
<div id="how-to-search-for-variants-within-a-certain-genomic-region" class="section level2">
<h2 class="hasAnchor">
<a href="#how-to-search-for-variants-within-a-certain-genomic-region" class="anchor"></a>5 | How to search for variants within a certain genomic region?</h2>
<div id="single-genomic-range" class="section level3">
<h3 class="hasAnchor">
<a href="#single-genomic-range" class="anchor"></a>Single genomic range</h3>
<p>For this you may use the function <code><a href="../reference/get_variants.html">get_variants()</a></code> with parameter <code>genomic_range</code>.</p>
<p>For example, to search for variants in chromosome Y in the interval 14692000â€“14695000, you start by defining a list of 3 elements: <code>chromosome</code>, <code>start</code> and <code>end</code> that specify your genomic range:</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" title="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(gwasrapidd)</a>
<a class="sourceLine" id="cb6-2" title="2"><span class="co"># 'chromosome' names are case sensitive, and should be uppercase.</span></a>
<a class="sourceLine" id="cb6-3" title="3"><span class="co"># 'start' and 'end' positions should be integer vectors.</span></a>
<a class="sourceLine" id="cb6-4" title="4">my_genomic_range &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(</a>
<a class="sourceLine" id="cb6-5" title="5">  <span class="dt">chromosome =</span> <span class="st">'Y'</span>,</a>
<a class="sourceLine" id="cb6-6" title="6">  <span class="dt">start =</span> 14692000L,</a>
<a class="sourceLine" id="cb6-7" title="7">  <span class="dt">end =</span> 14695000L)</a></code></pre></div>
<p>Now you can use <code>my_genomic_range</code> to retrieve the variants:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" title="1">chr_Y_variants &lt;-<span class="st"> </span><span class="kw"><a href="../reference/get_variants.html">get_variants</a></span>(<span class="dt">genomic_range =</span> my_genomic_range)</a>
<a class="sourceLine" id="cb7-2" title="2">chr_Y_variants<span class="op">@</span>variants[<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">'variant_id'</span>, <span class="st">'functional_class'</span>)]</a>
<a class="sourceLine" id="cb7-3" title="3"><span class="co">#&gt; # A tibble: 1 x 2</span></a>
<a class="sourceLine" id="cb7-4" title="4"><span class="co">#&gt;   variant_id functional_class</span></a>
<a class="sourceLine" id="cb7-5" title="5"><span class="co">#&gt;   &lt;chr&gt;      &lt;chr&gt;           </span></a>
<a class="sourceLine" id="cb7-6" title="6"><span class="co">#&gt; 1 rs2115848  intron_variant</span></a></code></pre></div>
</div>
<div id="multiple-genomic-ranges" class="section level3">
<h3 class="hasAnchor">
<a href="#multiple-genomic-ranges" class="anchor"></a>Multiple genomic ranges</h3>
<p>To search in multiple regions, construct your genomic range list with those locations just like in the previous example. For example, letâ€™s search now for variants in chromosome X and Y, both in range 13000000â€“15000000:</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" title="1">my_genomic_range &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(</a>
<a class="sourceLine" id="cb8-2" title="2">  <span class="dt">chromosome =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">'X'</span>, <span class="st">'Y'</span>),</a>
<a class="sourceLine" id="cb8-3" title="3">  <span class="dt">start =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(13000000L, 13000000L),</a>
<a class="sourceLine" id="cb8-4" title="4">  <span class="dt">end =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(15000000L, 15000000L))</a>
<a class="sourceLine" id="cb8-5" title="5"></a>
<a class="sourceLine" id="cb8-6" title="6">chr_XY_variants &lt;-<span class="st"> </span><span class="kw"><a href="../reference/get_variants.html">get_variants</a></span>(<span class="dt">genomic_range =</span> my_genomic_range)</a>
<a class="sourceLine" id="cb8-7" title="7">chr_XY_variants<span class="op">@</span>variants[<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">'variant_id'</span>,</a>
<a class="sourceLine" id="cb8-8" title="8">                           <span class="st">'chromosome_name'</span>,</a>
<a class="sourceLine" id="cb8-9" title="9">                           <span class="st">'chromosome_position'</span>)]</a>
<a class="sourceLine" id="cb8-10" title="10"><span class="co">#&gt; # A tibble: 7 x 3</span></a>
<a class="sourceLine" id="cb8-11" title="11"><span class="co">#&gt;   variant_id  chromosome_name chromosome_position</span></a>
<a class="sourceLine" id="cb8-12" title="12"><span class="co">#&gt;   &lt;chr&gt;       &lt;chr&gt;                         &lt;int&gt;</span></a>
<a class="sourceLine" id="cb8-13" title="13"><span class="co">#&gt; 1 rs5980075   X                          14932409</span></a>
<a class="sourceLine" id="cb8-14" title="14"><span class="co">#&gt; 2 rs6528024   X                          13912803</span></a>
<a class="sourceLine" id="cb8-15" title="15"><span class="co">#&gt; 3 rs850637    X                          13005622</span></a>
<a class="sourceLine" id="cb8-16" title="16"><span class="co">#&gt; 4 rs5978649   X                          13596999</span></a>
<a class="sourceLine" id="cb8-17" title="17"><span class="co">#&gt; 5 rs138331350 X                          14841981</span></a>
<a class="sourceLine" id="cb8-18" title="18"><span class="co">#&gt; 6 rs749624882 X                          14929148</span></a>
<a class="sourceLine" id="cb8-19" title="19"><span class="co">#&gt; 7 rs2115848   Y                          14692972</span></a></code></pre></div>
</div>
<div id="searching-variants-by-cytogenetic-regions" class="section level3">
<h3 class="hasAnchor">
<a href="#searching-variants-by-cytogenetic-regions" class="anchor"></a>Searching variants by cytogenetic regions</h3>
<p>To search for variants within a cytogenetic band you can use the parameter <code>cytogenetic_band</code> of <code><a href="../reference/get_variants.html">get_variants()</a></code>. Here is an example, again for chromosome Y, using the cytogenetic band <code>'Yq11.221'</code> as query:</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" title="1">my_variants &lt;-<span class="st"> </span><span class="kw"><a href="../reference/get_variants.html">get_variants</a></span>(<span class="dt">cytogenetic_band =</span> <span class="st">'Yq11.221'</span>)</a>
<a class="sourceLine" id="cb9-2" title="2">my_variants<span class="op">@</span>variants[<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">'variant_id'</span>,</a>
<a class="sourceLine" id="cb9-3" title="3">                       <span class="st">'chromosome_name'</span>,</a>
<a class="sourceLine" id="cb9-4" title="4">                       <span class="st">'chromosome_position'</span>)]</a>
<a class="sourceLine" id="cb9-5" title="5"><span class="co">#&gt; # A tibble: 1 x 3</span></a>
<a class="sourceLine" id="cb9-6" title="6"><span class="co">#&gt;   variant_id chromosome_name chromosome_position</span></a>
<a class="sourceLine" id="cb9-7" title="7"><span class="co">#&gt;   &lt;chr&gt;      &lt;chr&gt;                         &lt;int&gt;</span></a>
<a class="sourceLine" id="cb9-8" title="8"><span class="co">#&gt; 1 rs2115848  Y                          14692972</span></a></code></pre></div>
<p>How to know what are the cytogenetic bands for querying? We provide a dataset (dataframe) named <code>cytogenetic_bands</code> that you can use:</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" title="1"><span class="co"># ?cytogenetic_bands for more details.</span></a>
<a class="sourceLine" id="cb10-2" title="2">cytogenetic_bands</a>
<a class="sourceLine" id="cb10-3" title="3"><span class="co">#&gt; # A tibble: 862 x 8</span></a>
<a class="sourceLine" id="cb10-4" title="4"><span class="co">#&gt;    cytogenetic_band chromosome  start    end length assembly stain</span></a>
<a class="sourceLine" id="cb10-5" title="5"><span class="co">#&gt;    &lt;chr&gt;            &lt;chr&gt;       &lt;int&gt;  &lt;int&gt;  &lt;dbl&gt; &lt;chr&gt;    &lt;chr&gt;</span></a>
<a class="sourceLine" id="cb10-6" title="6"><span class="co">#&gt;  1 1p36.33          1          1.00e0 2.30e6 2.30e6 GRCh38   gneg </span></a>
<a class="sourceLine" id="cb10-7" title="7"><span class="co">#&gt;  2 1p36.32          1          2.30e6 5.30e6 3.00e6 GRCh38   gposâ€¦</span></a>
<a class="sourceLine" id="cb10-8" title="8"><span class="co">#&gt;  3 1p36.31          1          5.30e6 7.10e6 1.80e6 GRCh38   gneg </span></a>
<a class="sourceLine" id="cb10-9" title="9"><span class="co">#&gt;  4 1p36.23          1          7.10e6 9.10e6 2.00e6 GRCh38   gposâ€¦</span></a>
<a class="sourceLine" id="cb10-10" title="10"><span class="co">#&gt;  5 1p36.22          1          9.10e6 1.25e7 3.40e6 GRCh38   gneg </span></a>
<a class="sourceLine" id="cb10-11" title="11"><span class="co">#&gt;  6 1p36.21          1          1.25e7 1.59e7 3.40e6 GRCh38   gposâ€¦</span></a>
<a class="sourceLine" id="cb10-12" title="12"><span class="co">#&gt;  7 1p36.13          1          1.59e7 2.01e7 4.20e6 GRCh38   gneg </span></a>
<a class="sourceLine" id="cb10-13" title="13"><span class="co">#&gt;  8 1p36.12          1          2.01e7 2.36e7 3.50e6 GRCh38   gposâ€¦</span></a>
<a class="sourceLine" id="cb10-14" title="14"><span class="co">#&gt;  9 1p36.11          1          2.36e7 2.76e7 4.00e6 GRCh38   gneg </span></a>
<a class="sourceLine" id="cb10-15" title="15"><span class="co">#&gt; 10 1p35.3           1          2.76e7 2.99e7 2.30e6 GRCh38   gposâ€¦</span></a>
<a class="sourceLine" id="cb10-16" title="16"><span class="co">#&gt; # â€¦ with 852 more rows, and 1 more variable: last_download_date &lt;chr&gt;</span></a></code></pre></div>
<p>Letâ€™s say you want to search for all variants in the shorter arm (p) of chromosome 21, you can take advantage of the <code>cytogenetic_bands</code> to get all the corresponding cytogenetic band names:</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" title="1"><span class="co"># Install package dplyr if you do not have it.</span></a>
<a class="sourceLine" id="cb11-2" title="2">chr21_p_bands &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(cytogenetic_bands, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/grep">grepl</a></span>(<span class="st">'^21p'</span>, cytogenetic_band)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb11-3" title="3"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/pull.html">pull</a></span>(cytogenetic_band)</a>
<a class="sourceLine" id="cb11-4" title="4">chr21_p_bands</a>
<a class="sourceLine" id="cb11-5" title="5"><span class="co">#&gt; [1] "21p13"   "21p12"   "21p11.2" "21p11.1"</span></a></code></pre></div>
<p>Now search by <code>cytogenetic_band</code>:</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" title="1">my_variants &lt;-<span class="st"> </span><span class="kw"><a href="../reference/get_variants.html">get_variants</a></span>(<span class="dt">cytogenetic_band =</span> chr21_p_bands)</a>
<a class="sourceLine" id="cb12-2" title="2">my_variants<span class="op">@</span>variants[<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">'variant_id'</span>,</a>
<a class="sourceLine" id="cb12-3" title="3">                       <span class="st">'chromosome_name'</span>,</a>
<a class="sourceLine" id="cb12-4" title="4">                       <span class="st">'chromosome_position'</span>,</a>
<a class="sourceLine" id="cb12-5" title="5">                       <span class="st">'chromosome_region'</span>)]</a>
<a class="sourceLine" id="cb12-6" title="6"><span class="co">#&gt; # A tibble: 2 x 4</span></a>
<a class="sourceLine" id="cb12-7" title="7"><span class="co">#&gt;   variant_id chromosome_name chromosome_position chromosome_region</span></a>
<a class="sourceLine" id="cb12-8" title="8"><span class="co">#&gt;   &lt;chr&gt;      &lt;chr&gt;                         &lt;int&gt; &lt;chr&gt;            </span></a>
<a class="sourceLine" id="cb12-9" title="9"><span class="co">#&gt; 1 rs240444   21                         10510446 21p11.2          </span></a>
<a class="sourceLine" id="cb12-10" title="10"><span class="co">#&gt; 2 rs10439884 21                         10540506 21p11.2</span></a></code></pre></div>
</div>
</div>
<div id="genomic-range-for-an-entire-chromosome" class="section level2">
<h2 class="hasAnchor">
<a href="#genomic-range-for-an-entire-chromosome" class="anchor"></a>6 | Genomic range for an entire chromosome?</h2>
<p>You can get the total length of a chromosome by using the provided dataset: <code>cytogenetic_bands</code>. Here is an example for chromosome 15:</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" title="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(gwasrapidd)</a>
<a class="sourceLine" id="cb13-2" title="2"><span class="co"># Install dplyr first.</span></a>
<a class="sourceLine" id="cb13-3" title="3">dplyr<span class="op">::</span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(cytogenetic_bands, chromosome <span class="op">==</span><span class="st"> '15'</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb13-4" title="4"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/summarise.html">summarise</a></span>(<span class="dt">chromosome =</span> dplyr<span class="op">::</span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/nth.html">first</a></span>(chromosome),</a>
<a class="sourceLine" id="cb13-5" title="5">                   <span class="dt">start =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/Extremes">min</a></span>(start),</a>
<a class="sourceLine" id="cb13-6" title="6">                   <span class="dt">end =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/Extremes">max</a></span>(end)</a>
<a class="sourceLine" id="cb13-7" title="7">                   )</a>
<a class="sourceLine" id="cb13-8" title="8"><span class="co">#&gt; # A tibble: 1 x 3</span></a>
<a class="sourceLine" id="cb13-9" title="9"><span class="co">#&gt;   chromosome start       end</span></a>
<a class="sourceLine" id="cb13-10" title="10"><span class="co">#&gt;   &lt;chr&gt;      &lt;int&gt;     &lt;int&gt;</span></a>
<a class="sourceLine" id="cb13-11" title="11"><span class="co">#&gt; 1 15             1 101991189</span></a></code></pre></div>
</div>
<div id="how-to-keep-track-of-which-queries-generated-which-results" class="section level2">
<h2 class="hasAnchor">
<a href="#how-to-keep-track-of-which-queries-generated-which-results" class="anchor"></a>7 | How to keep track of which queries generated which results?</h2>
<p>Currently, there is not an implemented solution in gwasrapidd. For example, if you search for variants by EFO identifier (<code>efo_id</code>):</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" title="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(gwasrapidd)</a>
<a class="sourceLine" id="cb14-2" title="2">my_efo_ids &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">'EFO_0005543'</span>, <span class="st">'EFO_0004762'</span>)</a>
<a class="sourceLine" id="cb14-3" title="3">my_variants &lt;-<span class="st"> </span><span class="kw"><a href="../reference/get_variants.html">get_variants</a></span>(<span class="dt">efo_id =</span> my_efo_ids)</a>
<a class="sourceLine" id="cb14-4" title="4">my_variants<span class="op">@</span>variants<span class="op">$</span>variant_id</a>
<a class="sourceLine" id="cb14-5" title="5"><span class="co">#&gt;  [1] "rs7572263"   "rs3751667"   "rs11599775"  "rs12803321"  "rs2297440"  </span></a>
<a class="sourceLine" id="cb14-6" title="6"><span class="co">#&gt;  [6] "rs10069690"  "rs75061358"  "rs12752552"  "rs4252707"   "rs11598018" </span></a>
<a class="sourceLine" id="cb14-7" title="7"><span class="co">#&gt; [11] "rs11233250"  "rs10131032"  "rs10852606"  "rs2235573"   "rs3772190"  </span></a>
<a class="sourceLine" id="cb14-8" title="8"><span class="co">#&gt; [16] "rs634537"    "rs77633900"  "rs78378222"  "rs12076373"  "rs11706832" </span></a>
<a class="sourceLine" id="cb14-9" title="9"><span class="co">#&gt; [21] "rs7107785"   "rs648044"    "rs1275600"   "rs55705857"  "rs9841110"  </span></a>
<a class="sourceLine" id="cb14-10" title="10"><span class="co">#&gt; [26] "rs11979158"  "rs4513773"   "rs6993770"   "rs10738760"  "rs9472155"  </span></a>
<a class="sourceLine" id="cb14-11" title="11"><span class="co">#&gt; [31] "rs10761741"  "rs34528081"  "rs2639990"   "rs2375981"   "rs6921438"  </span></a>
<a class="sourceLine" id="cb14-12" title="12"><span class="co">#&gt; [36] "rs114694170" "rs1740073"   "rs4782371"   "rs7043199"   "rs7030781"  </span></a>
<a class="sourceLine" id="cb14-13" title="13"><span class="co">#&gt; [41] "rs139893147" "rs186066666" "rs150821445" "rs75455100"  "rs4082730"  </span></a>
<a class="sourceLine" id="cb14-14" title="14"><span class="co">#&gt; [46] "rs12445232"  "rs8045833"   "rs10153304"  "rs12456390"  "rs10411345" </span></a>
<a class="sourceLine" id="cb14-15" title="15"><span class="co">#&gt; [51] "rs11392719"  "rs61829244"  "rs10934631"  "rs180936035" "rs187918360"</span></a>
<a class="sourceLine" id="cb14-16" title="16"><span class="co">#&gt; [56] "rs181558074" "rs191332118" "rs73872715"  "rs77961527"  "rs143479231"</span></a>
<a class="sourceLine" id="cb14-17" title="17"><span class="co">#&gt; [61] "rs59706856"  "rs71779653"  "rs181216459" "rs12199215"  "rs7763358"  </span></a>
<a class="sourceLine" id="cb14-18" title="18"><span class="co">#&gt; [66] "rs144085478" "rs144820908" "rs117580153" "rs118086804" "rs144160960"</span></a>
<a class="sourceLine" id="cb14-19" title="19"><span class="co">#&gt; [71] "rs6479877"   "rs10761731"  "rs10761750"  "rs34524635"  "rs73418461" </span></a>
<a class="sourceLine" id="cb14-20" title="20"><span class="co">#&gt; [76] "rs10886366"  "rs34881325"  "rs12214617"</span></a></code></pre></div>
<p>So it is not immediately obvious which variants resulted from the query <code>'EFO_0005543'</code> or <code>'EFO_0004762'</code>.</p>
<p>A possible workaround is to make multiple independent queries and save your results in a list whose names are the respective queries:</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" title="1"><span class="co"># Install purrr first.</span></a>
<a class="sourceLine" id="cb15-2" title="2"><span class="co"># Add names to my_efo_ids</span></a>
<a class="sourceLine" id="cb15-3" title="3"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/names">names</a></span>(my_efo_ids) &lt;-<span class="st"> </span>my_efo_ids</a>
<a class="sourceLine" id="cb15-4" title="4">my_variants &lt;-<span class="st"> </span>purrr<span class="op">::</span><span class="kw"><a href="https://purrr.tidyverse.org/reference/map.html">map</a></span>(my_efo_ids, <span class="op">~</span><span class="st"> </span><span class="kw"><a href="../reference/get_variants.html">get_variants</a></span>(<span class="dt">efo_id =</span> .x))</a></code></pre></div>
<p>Now you can see which variants are associated with each EFO identifier.</p>
<p>For <code>'EFO_0005543'</code> we got the following variants:</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" title="1">my_variants[[<span class="st">'EFO_0005543'</span>]]<span class="op">@</span>variants<span class="op">$</span>variant_id</a>
<a class="sourceLine" id="cb16-2" title="2"><span class="co">#&gt;  [1] "rs7572263"  "rs3751667"  "rs11599775" "rs12803321" "rs2297440" </span></a>
<a class="sourceLine" id="cb16-3" title="3"><span class="co">#&gt;  [6] "rs10069690" "rs75061358" "rs12752552" "rs4252707"  "rs11598018"</span></a>
<a class="sourceLine" id="cb16-4" title="4"><span class="co">#&gt; [11] "rs11233250" "rs10131032" "rs10852606" "rs2235573"  "rs3772190" </span></a>
<a class="sourceLine" id="cb16-5" title="5"><span class="co">#&gt; [16] "rs634537"   "rs77633900" "rs78378222" "rs12076373" "rs11706832"</span></a>
<a class="sourceLine" id="cb16-6" title="6"><span class="co">#&gt; [21] "rs7107785"  "rs648044"   "rs1275600"  "rs55705857" "rs9841110" </span></a>
<a class="sourceLine" id="cb16-7" title="7"><span class="co">#&gt; [26] "rs11979158"</span></a></code></pre></div>
<p>And for <code>'EFO_0004762'</code>:</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" title="1">my_variants[[<span class="st">'EFO_0004762'</span>]]<span class="op">@</span>variants<span class="op">$</span>variant_id</a>
<a class="sourceLine" id="cb17-2" title="2"><span class="co">#&gt;  [1] "rs4513773"   "rs6993770"   "rs10738760"  "rs9472155"   "rs10761741" </span></a>
<a class="sourceLine" id="cb17-3" title="3"><span class="co">#&gt;  [6] "rs34528081"  "rs2639990"   "rs2375981"   "rs6921438"   "rs114694170"</span></a>
<a class="sourceLine" id="cb17-4" title="4"><span class="co">#&gt; [11] "rs1740073"   "rs4782371"   "rs7043199"   "rs7030781"   "rs139893147"</span></a>
<a class="sourceLine" id="cb17-5" title="5"><span class="co">#&gt; [16] "rs186066666" "rs150821445" "rs75455100"  "rs4082730"   "rs12445232" </span></a>
<a class="sourceLine" id="cb17-6" title="6"><span class="co">#&gt; [21] "rs8045833"   "rs10153304"  "rs12456390"  "rs10411345"  "rs11392719" </span></a>
<a class="sourceLine" id="cb17-7" title="7"><span class="co">#&gt; [26] "rs61829244"  "rs10934631"  "rs180936035" "rs187918360" "rs181558074"</span></a>
<a class="sourceLine" id="cb17-8" title="8"><span class="co">#&gt; [31] "rs191332118" "rs73872715"  "rs77961527"  "rs143479231" "rs59706856" </span></a>
<a class="sourceLine" id="cb17-9" title="9"><span class="co">#&gt; [36] "rs71779653"  "rs181216459" "rs12199215"  "rs7763358"   "rs144085478"</span></a>
<a class="sourceLine" id="cb17-10" title="10"><span class="co">#&gt; [41] "rs144820908" "rs117580153" "rs118086804" "rs144160960" "rs6479877"  </span></a>
<a class="sourceLine" id="cb17-11" title="11"><span class="co">#&gt; [46] "rs10761731"  "rs10761750"  "rs34524635"  "rs73418461"  "rs10886366" </span></a>
<a class="sourceLine" id="cb17-12" title="12"><span class="co">#&gt; [51] "rs34881325"  "rs12214617"</span></a></code></pre></div>
</div>
<div id="how-to-combine-results-from-multiple-queries" class="section level2">
<h2 class="hasAnchor">
<a href="#how-to-combine-results-from-multiple-queries" class="anchor"></a>8 | How to combine results from multiple queries?</h2>
<p>The four main retrieval functions <code><a href="../reference/get_studies.html">get_studies()</a></code>, <code><a href="../reference/get_associations.html">get_associations()</a></code>, <code><a href="../reference/get_variants.html">get_variants()</a></code> and <code><a href="../reference/get_traits.html">get_traits()</a></code> all allow you to search multiple criteria at once. You can then combine results in an OR or AND fashion using the parameter <code>set_operation</code>.</p>
<p>Use <code>set_operation = 'union'</code> to combine results in an OR fashion:</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb18-1" title="1">my_variants_OR &lt;-<span class="st"> </span><span class="kw"><a href="../reference/get_variants.html">get_variants</a></span>(</a>
<a class="sourceLine" id="cb18-2" title="2">  <span class="dt">efo_trait =</span> <span class="st">'triple-negative breast cancer'</span>,</a>
<a class="sourceLine" id="cb18-3" title="3">  <span class="dt">gene_name =</span> <span class="st">'MDM4'</span>,</a>
<a class="sourceLine" id="cb18-4" title="4">  <span class="dt">set_operation =</span> <span class="st">'union'</span>)</a>
<a class="sourceLine" id="cb18-5" title="5"></a>
<a class="sourceLine" id="cb18-6" title="6">my_variants_OR<span class="op">@</span>variants[<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">'variant_id'</span>,</a>
<a class="sourceLine" id="cb18-7" title="7">                       <span class="st">'chromosome_name'</span>,</a>
<a class="sourceLine" id="cb18-8" title="8">                       <span class="st">'chromosome_position'</span>,</a>
<a class="sourceLine" id="cb18-9" title="9">                       <span class="st">'chromosome_region'</span>)]</a>
<a class="sourceLine" id="cb18-10" title="10"><span class="co">#&gt; # A tibble: 24 x 4</span></a>
<a class="sourceLine" id="cb18-11" title="11"><span class="co">#&gt;    variant_id chromosome_name chromosome_position chromosome_region</span></a>
<a class="sourceLine" id="cb18-12" title="12"><span class="co">#&gt;    &lt;chr&gt;      &lt;chr&gt;                         &lt;int&gt; &lt;chr&gt;            </span></a>
<a class="sourceLine" id="cb18-13" title="13"><span class="co">#&gt;  1 rs3789045  1                         204617684 1q32.1           </span></a>
<a class="sourceLine" id="cb18-14" title="14"><span class="co">#&gt;  2 rs2290854  1                         204546897 1q32.1           </span></a>
<a class="sourceLine" id="cb18-15" title="15"><span class="co">#&gt;  3 rs3789044  1                         204619973 1q32.1           </span></a>
<a class="sourceLine" id="cb18-16" title="16"><span class="co">#&gt;  4 rs1980050  1                         204477560 1q32.1           </span></a>
<a class="sourceLine" id="cb18-17" title="17"><span class="co">#&gt;  5 rs7551222  1                         204630167 1q32.1           </span></a>
<a class="sourceLine" id="cb18-18" title="18"><span class="co">#&gt;  6 rs2137255  1                         204457245 1q32.1           </span></a>
<a class="sourceLine" id="cb18-19" title="19"><span class="co">#&gt;  7 rs4252707  1                         204539019 1q32.1           </span></a>
<a class="sourceLine" id="cb18-20" title="20"><span class="co">#&gt;  8 rs11584700 1                         204607855 1q32.1           </span></a>
<a class="sourceLine" id="cb18-21" title="21"><span class="co">#&gt;  9 rs4951404  1                         204572214 1q32.1           </span></a>
<a class="sourceLine" id="cb18-22" title="22"><span class="co">#&gt; 10 rs16853958 1                         204559216 1q32.1           </span></a>
<a class="sourceLine" id="cb18-23" title="23"><span class="co">#&gt; # â€¦ with 14 more rows</span></a></code></pre></div>
<p>Or use <code>set_operation = 'intersection'</code> to combine results in an AND fashion:</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb19-1" title="1">my_variants_AND &lt;-<span class="st"> </span><span class="kw"><a href="../reference/get_variants.html">get_variants</a></span>(</a>
<a class="sourceLine" id="cb19-2" title="2">  <span class="dt">efo_trait =</span> <span class="st">'triple-negative breast cancer'</span>,</a>
<a class="sourceLine" id="cb19-3" title="3">  <span class="dt">gene_name =</span> <span class="st">'MDM4'</span>,</a>
<a class="sourceLine" id="cb19-4" title="4">  <span class="dt">set_operation =</span> <span class="st">'intersection'</span>)</a>
<a class="sourceLine" id="cb19-5" title="5"></a>
<a class="sourceLine" id="cb19-6" title="6">my_variants_AND<span class="op">@</span>variants[<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">'variant_id'</span>,</a>
<a class="sourceLine" id="cb19-7" title="7">                       <span class="st">'chromosome_name'</span>,</a>
<a class="sourceLine" id="cb19-8" title="8">                       <span class="st">'chromosome_position'</span>,</a>
<a class="sourceLine" id="cb19-9" title="9">                       <span class="st">'chromosome_region'</span>)]</a>
<a class="sourceLine" id="cb19-10" title="10"><span class="co">#&gt; # A tibble: 1 x 4</span></a>
<a class="sourceLine" id="cb19-11" title="11"><span class="co">#&gt;   variant_id chromosome_name chromosome_position chromosome_region</span></a>
<a class="sourceLine" id="cb19-12" title="12"><span class="co">#&gt;   &lt;chr&gt;      &lt;chr&gt;                         &lt;int&gt; &lt;chr&gt;            </span></a>
<a class="sourceLine" id="cb19-13" title="13"><span class="co">#&gt; 1 rs4245739  1                         204549714 1q32.1</span></a></code></pre></div>
<!--
## 9 | How to exclude results for specific queries?

The approach here is to search first using all your queries, including those queries for results you want to exclude, and then remove them with `setdiff`.

Let's get all variants in the cytogenetic region

## 5 | How to search by EFO identifier, including its child terms?

The [Experimental Factor Ontology (EFO)](https://www.ebi.ac.uk/efo/) identifiers can have one or more child terms as part of its EFO hierarchy. These child terms have a narrower meaning, and oftentimes one would like to search by an EFO identifier and also by all its child identifiers. This functionality is provided by the [GWAS Catalog Web User Interface](https://www.ebi.ac.uk/gwas/) in a check box, see image below.

<img src="../man/figures/child_trait.png" style="display: block; margin: auto auto auto 0;" />

However there is no such option for the REST API. To circumvent this problem you can use the function `get_child_efo()` that will get all child terms of a given EFO identifier. Here's an example with the EFO identifier `'EFO_0005543'` (glioma):



```r
library(gwasrapidd)
my_efo_id <- 'EFO_0005543'
my_studies <- get_studies(efo_id = my_efo_id)
n(my_studies)
#> [1] 5
```


```r
child_efo_ids <- get_child_efo(efo_id = my_efo_id)
my_studies_w_children <- get_studies(efo_id = c(my_efo_id, child_efo_ids[['my_efo_id']]))
n(my_studies_w_children)
#> [1] 5
```

We find 5 studies using `'EFO_0005543'` only, and 5 studies if we include also the child EFO identifiers.
-->
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#how-to-be-sure-that-i-can-establish-a-connection-to-the-gwas-catalog-server">1 | How to be sure that I can establish a connection to the GWAS Catalog server?</a></li>
      <li><a href="#what-resources-is-the-gwas-catalog-database-currently-mapped-against">2 | What resources is the GWAS Catalog database currently mapped against?</a></li>
      <li><a href="#what-is-the-difference-between-a-trait-and-a-reported-trait">3 | What is the difference between a trait and a reported trait?</a></li>
      <li><a href="#genomic-coordinates-of-genomic-contexts-seem-to-be-wrong">4 | Genomic coordinates of genomic contexts seem to be wrong?</a></li>
      <li><a href="#how-to-search-for-variants-within-a-certain-genomic-region">5 | How to search for variants within a certain genomic region?</a></li>
      <li><a href="#genomic-range-for-an-entire-chromosome">6 | Genomic range for an entire chromosome?</a></li>
      <li><a href="#how-to-keep-track-of-which-queries-generated-which-results">7 | How to keep track of which queries generated which results?</a></li>
      <li><a href="#how-to-combine-results-from-multiple-queries">8 | How to combine results from multiple queries?</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Ramiro Magno, Ana-Teresa Maia, Centre for Biomedical Research, University of Algarve.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
