<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Getting started with gwasrapidd • gwasrapidd</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Getting started with gwasrapidd">
<meta property="og:description" content="">
<meta property="og:image" content="https://rmagno.eu/gwasrapidd/logo.png">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">gwasrapidd</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.7</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/gwasrapidd.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/faq.html">Frequently asked questions</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/ramiromagno/gwasrapidd">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Getting started with gwasrapidd</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/ramiromagno/gwasrapidd/blob/master/vignettes/gwasrapidd.Rmd"><code>vignettes/gwasrapidd.Rmd</code></a></small>
      <div class="hidden name"><code>gwasrapidd.Rmd</code></div>

    </div>

    
    
<div id="the-gwas-catalog" class="section level2">
<h2 class="hasAnchor">
<a href="#the-gwas-catalog" class="anchor"></a>The GWAS Catalog</h2>
<p>The GWAS Catalog is a service provided by the EMBL-EBI and NHGRI that offers a manually curated and freely available database of published genome-wide association studies (GWAS). The Catalog website and infrastructure is hosted by the <a href="https://www.ebi.ac.uk">EMBL-EBI</a>.</p>
<p>There are three ways to access the Catalog database:</p>
<ul>
<li>Via the Web interface at <a href="https://www.ebi.ac.uk/gwas">https://www.ebi.ac.uk/gwas</a>
</li>
<li>By downloading a database dump in TSV or RDF/OWL formats: <a href="https://www.ebi.ac.uk/gwas/docs/file-downloads">https://www.ebi.ac.uk/gwas/docs/file-downloads</a>
</li>
<li>Or through the GWAS Catalog REST API: <a href="https://www.ebi.ac.uk/gwas/rest/docs/api">https://www.ebi.ac.uk/gwas/rest/docs/api</a>
</li>
</ul>
<p>gwasrapidd facilitates the access to the Catalog via the REST API, allowing you to programmatically retrieve data directly into R.</p>
</div>
<div id="gwas-catalog-entities" class="section level2">
<h2 class="hasAnchor">
<a href="#gwas-catalog-entities" class="anchor"></a>GWAS Catalog Entities</h2>
<p>The Catalog REST API is organized around four core entities: studies, associations, variants, and traits. <strong>gwasrapidd</strong> provides four corresponding functions to get each of the entities: <code><a href="../reference/get_studies.html">get_studies()</a></code>, <code><a href="../reference/get_associations.html">get_associations()</a></code>, <code><a href="../reference/get_variants.html">get_variants()</a></code>, and <code><a href="../reference/get_traits.html">get_traits()</a></code>.</p>
<p>Each function maps to an appropriately named S4 classed object: <a href="https://rmagno.eu/gwasrapidd/reference/studies-class.html">studies</a>, <a href="https://rmagno.eu/gwasrapidd/reference/associations-class.html">associations</a>, <a href="https://rmagno.eu/gwasrapidd/reference/variants-class.html">variants</a>, and <a href="https://rmagno.eu/gwasrapidd/reference/traits-class.html">traits</a> (see Figure 1).</p>
<div class="figure" style="text-align: left">
<img src="../man/figures/get_fns.png" alt="Figure 1 | gwasrapidd retrieval functions." width="500"><p class="caption">
Figure 1 | gwasrapidd retrieval functions.
</p>
</div>
<p>You can use a combination of several search criteria with each retrieval function as shown in Figure 2. For example, if you want to get studies using either one of these two criteria: <em>study accession identifier</em> and <em>variant identifier</em>, you could run the following code:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" title="1">my_studies &lt;-<span class="st"> </span><span class="kw"><a href="../reference/get_studies.html">get_studies</a></span>(<span class="dt">study_id =</span> <span class="st">'GCST000858'</span>, <span class="dt">variant_id =</span> <span class="st">'rs12752552'</span>)</a></code></pre></div>
<p>This command returns all studies that match either <code>'GCST000858'</code> or <code>'rs12752552'</code>. This is equivalent to running <code>get_studies</code> separately on each criteria, and combining the results afterwards:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" title="1">s1 &lt;-<span class="st"> </span><span class="kw"><a href="../reference/get_studies.html">get_studies</a></span>(<span class="dt">study_id =</span> <span class="st">'GCST000858'</span>)</a>
<a class="sourceLine" id="cb2-2" title="2">s2 &lt;-<span class="st"> </span><span class="kw"><a href="../reference/get_studies.html">get_studies</a></span>(<span class="dt">variant_id =</span> <span class="st">'rs12752552'</span>)</a>
<a class="sourceLine" id="cb2-3" title="3">my_studies &lt;-<span class="st"> </span><span class="kw"><a href="../reference/setop.html">union</a></span>(s1, s2)</a></code></pre></div>
<p>All four retrieval functions accept the <code>set_operation</code> parameter which defines the way the results obtained with each criterion are combined. The two options for this parameter are <code>'union'</code> (default) or <code>'intersection'</code>, resulting, respectively, in an OR or AND operation.</p>
<div class="figure" style="text-align: left">
<img src="../man/figures/get_criteria.png" alt="Figure 2 | gwasrapidd arguments for retrieval functions. Colors indicate the criteria that can be used for retrieving GWAS Catalog entities: studies (green), associations (red), variants (purple), and traits (orange)." width="500"><p class="caption">
Figure 2 | gwasrapidd arguments for retrieval functions. Colors indicate the criteria that can be used for retrieving GWAS Catalog entities: studies (green), associations (red), variants (purple), and traits (orange).
</p>
</div>
</div>
<div id="example-1-finding-risk-alleles-associated-with-autoimmune-disease" class="section level2">
<h2 class="hasAnchor">
<a href="#example-1-finding-risk-alleles-associated-with-autoimmune-disease" class="anchor"></a>Example 1 | Finding Risk Alleles Associated with Autoimmune Disease</h2>
<p>As a first example, take the work by <span class="citation">Light et al. (2014)</span>. In this work the authors focused on variants that had been previously reported in genome-wide association studies (GWAS) for autoimmune disease.</p>
<p>With <strong>gwasrapidd</strong> we can interrogate the GWAS Catalog for the study/studies by searching by <em>autoimmune disease</em> (an EFO trait). To do that let’s load gwasrapidd first:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" title="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(gwasrapidd)</a></code></pre></div>
<p>Then query the GWAS Catalog by EFO trait:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" title="1">my_studies &lt;-<span class="st"> </span><span class="kw"><a href="../reference/get_studies.html">get_studies</a></span>(<span class="dt">efo_trait =</span> <span class="st">'autoimmune disease'</span>)</a></code></pre></div>
<p>We can now check how many GWAS studies we got back:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" title="1"><span class="kw"><a href="../reference/n.html">n</a></span>(my_studies)</a>
<a class="sourceLine" id="cb5-2" title="2"><span class="co">#&gt; [1] 1</span></a>
<a class="sourceLine" id="cb5-3" title="3">my_studies<span class="op">@</span>studies<span class="op">$</span>study_id</a>
<a class="sourceLine" id="cb5-4" title="4"><span class="co">#&gt; [1] "GCST003097"</span></a></code></pre></div>
<p>Apparently only one (GCST003097). Let’s see the associated publication title:</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" title="1">my_studies<span class="op">@</span>publications<span class="op">$</span>title</a>
<a class="sourceLine" id="cb6-2" title="2"><span class="co">#&gt; [1] "Meta-analysis of shared genetic architecture across ten pediatric autoimmune diseases."</span></a></code></pre></div>
<p>Sounds interesting… To quickly browse to the PubMed entry for this publication:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" title="1"><span class="co"># This launches your web browser at https://www.ncbi.nlm.nih.gov/pubmed/26301688</span></a>
<a class="sourceLine" id="cb7-2" title="2"><span class="kw"><a href="../reference/open_in_pubmed.html">open_in_pubmed</a></span>(my_studies<span class="op">@</span>publications<span class="op">$</span>pubmed_id) </a></code></pre></div>
<p>Now if we want to know the variants previously associated with autoimmune disease, as used by <span class="citation">Light et al. (2014)</span>, we need to retrieve statistical association information on these variants, and then filter them based on the same level of significance <span class="math inline">\(P &lt; 1\times 10^{-6}\)</span> <span class="citation">(Light et al. 2014)</span>.</p>
<p>So let’s start by getting the associations by <code>study_id</code>:</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" title="1"><span class="co"># You could have also used get_associations(efo_trait = 'autoimmune disease')</span></a>
<a class="sourceLine" id="cb8-2" title="2">my_associations &lt;-<span class="st"> </span><span class="kw"><a href="../reference/get_associations.html">get_associations</a></span>(<span class="dt">study_id =</span> <span class="st">"GCST003097"</span>)</a></code></pre></div>
<p>Seemingly, there are 46 associations.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" title="1"><span class="kw"><a href="../reference/n.html">n</a></span>(my_associations)</a>
<a class="sourceLine" id="cb9-2" title="2"><span class="co">#&gt; [1] 46</span></a></code></pre></div>
<p>However, not all variants meet the level of significance, as required by <span class="citation">Light et al. (2014)</span>:</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" title="1"><span class="co"># Get association ids for which pvalue is less than 1e-6.</span></a>
<a class="sourceLine" id="cb10-2" title="2">dplyr<span class="op">::</span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(my_associations<span class="op">@</span>associations, pvalue <span class="op">&lt;</span><span class="st"> </span><span class="fl">1e-6</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="co"># Filter by p-value</span></a>
<a class="sourceLine" id="cb10-3" title="3"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/pull.html">pull</a></span>(association_id) -&gt;<span class="st"> </span>association_ids <span class="co"># Extract column association_id</span></a></code></pre></div>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" title="1"><span class="co"># Extract associations by association id</span></a>
<a class="sourceLine" id="cb11-2" title="2">my_associations2 &lt;-<span class="st"> </span>my_associations[association_ids]</a>
<a class="sourceLine" id="cb11-3" title="3"><span class="kw"><a href="../reference/n.html">n</a></span>(my_associations2)</a>
<a class="sourceLine" id="cb11-4" title="4"><span class="co">#&gt; [1] 44</span></a></code></pre></div>
<p>Of the 46 associations found in GWAS Catalog, 44 meet the p-value threshold of <span class="math inline">\(1\times 10^{-6}\)</span>. Here are the variants, and their respective risk allele and risk frequency:</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" title="1">my_associations2<span class="op">@</span>risk_alleles[<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">'variant_id'</span>, <span class="st">'risk_allele'</span>, <span class="st">'risk_frequency'</span>)] <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb12-2" title="2"><span class="st">  </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/print">print</a></span>(<span class="dt">n =</span> <span class="ot">Inf</span>)</a>
<a class="sourceLine" id="cb12-3" title="3"><span class="co">#&gt; # A tibble: 44 x 3</span></a>
<a class="sourceLine" id="cb12-4" title="4"><span class="co">#&gt;    variant_id  risk_allele risk_frequency</span></a>
<a class="sourceLine" id="cb12-5" title="5"><span class="co">#&gt;    &lt;chr&gt;       &lt;chr&gt;                &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb12-6" title="6"><span class="co">#&gt;  1 rs2066363   C                    0.34 </span></a>
<a class="sourceLine" id="cb12-7" title="7"><span class="co">#&gt;  2 rs114846446 A                    0.01 </span></a>
<a class="sourceLine" id="cb12-8" title="8"><span class="co">#&gt;  3 rs7672495   C                    0.18 </span></a>
<a class="sourceLine" id="cb12-9" title="9"><span class="co">#&gt;  4 rs7660520   A                    0.26 </span></a>
<a class="sourceLine" id="cb12-10" title="10"><span class="co">#&gt;  5 rs7831697   G                    0.25 </span></a>
<a class="sourceLine" id="cb12-11" title="11"><span class="co">#&gt;  6 rs7042370   T                    0.43 </span></a>
<a class="sourceLine" id="cb12-12" title="12"><span class="co">#&gt;  7 rs10988542  C                    0.08 </span></a>
<a class="sourceLine" id="cb12-13" title="13"><span class="co">#&gt;  8 rs7100025   G                    0.34 </span></a>
<a class="sourceLine" id="cb12-14" title="14"><span class="co">#&gt;  9 rs77150043  T                    0.23 </span></a>
<a class="sourceLine" id="cb12-15" title="15"><span class="co">#&gt; 10 rs2807264   C                    0.21 </span></a>
<a class="sourceLine" id="cb12-16" title="16"><span class="co">#&gt; 11 rs12863738  T                    0.17 </span></a>
<a class="sourceLine" id="cb12-17" title="17"><span class="co">#&gt; 12 rs11580078  G                    0.43 </span></a>
<a class="sourceLine" id="cb12-18" title="18"><span class="co">#&gt; 13 rs6679677   A                    0.09 </span></a>
<a class="sourceLine" id="cb12-19" title="19"><span class="co">#&gt; 14 rs34884278  C                    0.3  </span></a>
<a class="sourceLine" id="cb12-20" title="20"><span class="co">#&gt; 15 rs6689858   C                    0.290</span></a>
<a class="sourceLine" id="cb12-21" title="21"><span class="co">#&gt; 16 rs2075184   T                    0.23 </span></a>
<a class="sourceLine" id="cb12-22" title="22"><span class="co">#&gt; 17 rs36001488  C                    0.48 </span></a>
<a class="sourceLine" id="cb12-23" title="23"><span class="co">#&gt; 18 rs4676410   A                    0.19 </span></a>
<a class="sourceLine" id="cb12-24" title="24"><span class="co">#&gt; 19 rs4625      G                    0.31 </span></a>
<a class="sourceLine" id="cb12-25" title="25"><span class="co">#&gt; 20 rs7725052   C                    0.43 </span></a>
<a class="sourceLine" id="cb12-26" title="26"><span class="co">#&gt; 21 rs7731626   A                    0.39 </span></a>
<a class="sourceLine" id="cb12-27" title="27"><span class="co">#&gt; 22 rs755374    T                    0.32 </span></a>
<a class="sourceLine" id="cb12-28" title="28"><span class="co">#&gt; 23 rs36051895  T                    0.290</span></a>
<a class="sourceLine" id="cb12-29" title="29"><span class="co">#&gt; 24 rs4246905   T                    0.28 </span></a>
<a class="sourceLine" id="cb12-30" title="30"><span class="co">#&gt; 25 rs706778    T                    0.41 </span></a>
<a class="sourceLine" id="cb12-31" title="31"><span class="co">#&gt; 26 rs10822050  C                    0.39 </span></a>
<a class="sourceLine" id="cb12-32" title="32"><span class="co">#&gt; 27 rs17885785  T                    0.2  </span></a>
<a class="sourceLine" id="cb12-33" title="33"><span class="co">#&gt; 28 rs17466626  G                    0.02 </span></a>
<a class="sourceLine" id="cb12-34" title="34"><span class="co">#&gt; 29 rs1689510   C                    0.31 </span></a>
<a class="sourceLine" id="cb12-35" title="35"><span class="co">#&gt; 30 rs12598357  G                    0.39 </span></a>
<a class="sourceLine" id="cb12-36" title="36"><span class="co">#&gt; 31 rs12928404  C                    0.38 </span></a>
<a class="sourceLine" id="cb12-37" title="37"><span class="co">#&gt; 32 rs12232497  C                    0.45 </span></a>
<a class="sourceLine" id="cb12-38" title="38"><span class="co">#&gt; 33 rs62131887  T                    0.28 </span></a>
<a class="sourceLine" id="cb12-39" title="39"><span class="co">#&gt; 34 rs2836882   A                    0.27 </span></a>
<a class="sourceLine" id="cb12-40" title="40"><span class="co">#&gt; 35 rs62324212  A                    0.42 </span></a>
<a class="sourceLine" id="cb12-41" title="41"><span class="co">#&gt; 36 rs4869313   T                    0.42 </span></a>
<a class="sourceLine" id="cb12-42" title="42"><span class="co">#&gt; 37 rs11741255  A                    0.42 </span></a>
<a class="sourceLine" id="cb12-43" title="43"><span class="co">#&gt; 38 rs11145763  C                    0.4  </span></a>
<a class="sourceLine" id="cb12-44" title="44"><span class="co">#&gt; 39 rs1250563   C                    0.290</span></a>
<a class="sourceLine" id="cb12-45" title="45"><span class="co">#&gt; 40 rs1332099   T                    0.46 </span></a>
<a class="sourceLine" id="cb12-46" title="46"><span class="co">#&gt; 41 rs72743477  G                    0.21 </span></a>
<a class="sourceLine" id="cb12-47" title="47"><span class="co">#&gt; 42 rs117372389 T                    0.02 </span></a>
<a class="sourceLine" id="cb12-48" title="48"><span class="co">#&gt; 43 rs602662    G                    0.49 </span></a>
<a class="sourceLine" id="cb12-49" title="49"><span class="co">#&gt; 44 rs2738774   A                    0.32</span></a></code></pre></div>
</div>
<div id="references" class="section level2 unnumbered">
<h2 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h2>
<div id="refs" class="references">
<div id="ref-Light2014">
<p>Light, Nicholas, Véronique Adoue, Bing Ge, Shu-Huang Chen, Tony Kwan, and Tomi Pastinen. 2014. “Interrogation of Allelic Chromatin States in Human Cells by High-Density Chip-Genotyping.” <em>Epigenetics</em> 9 (9): 1238–51.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#the-gwas-catalog">The GWAS Catalog</a></li>
      <li><a href="#gwas-catalog-entities">GWAS Catalog Entities</a></li>
      <li><a href="#example-1-finding-risk-alleles-associated-with-autoimmune-disease">Example 1 | Finding Risk Alleles Associated with Autoimmune Disease</a></li>
      <li><a href="#references">References</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Ramiro Magno, Ana-Teresa Maia, Centre for Biomedical Research, University of Algarve.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
