<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Subset a variants object — subset-variants • gwasrapidd</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />
<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous" />
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



  <link href="../extra.css" rel="stylesheet">
  
<meta property="og:title" content="Subset a variants object — subset-variants" />

<meta property="og:description" content="You can subset variants by identifier or by position using the
`[` operator." />

<meta property="og:image" content="https://rmagno.eu/gwasrapidd/logo.png" />
<meta name="twitter:card" content="summary" />



<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">gwasrapidd</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.7</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/gwasrapidd.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/faq.html">Frequently asked questions</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/ramiromagno/gwasrapidd">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Subset a variants object</h1>
    <small class="dont-index">Source: <a href='https://github.com/ramiromagno/gwasrapidd/blob/master/R/generics.R'><code>R/generics.R</code></a></small>
    <div class="hidden name"><code>subset-variants.Rd</code></div>
    </div>

    <div class="ref-description">
    
    <p>You can subset <a href='variants-class.html'>variants</a> by identifier or by position using the
<code>`[`</code> operator.</p>
    
    </div>

    <pre class="usage"># S4 method for variants,missing,missing,missing
[(x, i, j, ...,
  drop = FALSE)

# S4 method for variants,numeric,missing,missing
[(x, i, j, ...,
  drop = FALSE)

# S4 method for variants,character,missing,missing
[(x, i, j, ...,
  drop = FALSE)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>x</th>
      <td><p>A <a href='variants-class.html'>variants</a> object.</p></td>
    </tr>
    <tr>
      <th>i</th>
      <td><p>Position of the identifier or the name of the identifier itself.</p></td>
    </tr>
    <tr>
      <th>j</th>
      <td><p>Not used.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>Additional arguments not used here.</p></td>
    </tr>
    <tr>
      <th>drop</th>
      <td><p>Not used.</p></td>
    </tr>
    </table>
    
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A <a href='variants-class.html'>variants</a> object.</p>
    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'># Subset a variants object by identifier</span>
<span class='no'>variants_ex01</span>[<span class='st'>'rs4725504'</span>]</div><div class='output co'>#&gt; An object of class "variants"
#&gt; Slot "variants":
#&gt; <span style='color: #949494;'># A tibble: 1 x 7</span><span>
#&gt;   variant_id merged functional_class chromosome_name chromosome_posi…
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>       </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>            </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>                      </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span>
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> rs4725504       0 intergenic_vari… 7                      153</span><span style='text-decoration: underline;'>608</span><span>338
#&gt; </span><span style='color: #949494;'># … with 2 more variables: chromosome_region </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span style='color: #949494;'>, last_update_date </span><span style='color: #949494;font-style: italic;'>&lt;dttm&gt;</span><span>
#&gt; 
#&gt; Slot "genomic_contexts":
#&gt; </span><span style='color: #949494;'># A tibble: 4 x 11</span><span>
#&gt;   variant_id gene_name chromosome_name chromosome_posi… distance is_closest_gene
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>      </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>     </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>                      </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span>    </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span>          
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> rs4725504  DPP6      7                      153</span><span style='text-decoration: underline;'>608</span><span>338   </span><span style='text-decoration: underline;'>278</span><span>759 TRUE           
#&gt; </span><span style='color: #BCBCBC;'>2</span><span> rs4725504  DPP6      7                      153</span><span style='text-decoration: underline;'>608</span><span>338   </span><span style='text-decoration: underline;'>139</span><span>795 TRUE           
#&gt; </span><span style='color: #BCBCBC;'>3</span><span> rs4725504  PAXBP1P1  7                      153</span><span style='text-decoration: underline;'>608</span><span>338   </span><span style='text-decoration: underline;'>158</span><span>463 TRUE           
#&gt; </span><span style='color: #BCBCBC;'>4</span><span> rs4725504  LOC10798… 7                      153</span><span style='text-decoration: underline;'>608</span><span>338    </span><span style='text-decoration: underline;'>84</span><span>107 TRUE           
#&gt; </span><span style='color: #949494;'># … with 5 more variables: is_intergenic </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span style='color: #949494;'>, is_upstream </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span style='color: #949494;'>,
#&gt; #   is_downstream </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span style='color: #949494;'>, source </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span style='color: #949494;'>, mapping_method </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>
#&gt; 
#&gt; Slot "ensembl_ids":
#&gt; </span><span style='color: #949494;'># A tibble: 2 x 3</span><span>
#&gt;   variant_id gene_name ensembl_id     
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>      </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>     </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>          
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> rs4725504  DPP6      ENSG00000130226
#&gt; </span><span style='color: #BCBCBC;'>2</span><span> rs4725504  PAXBP1P1  ENSG00000233489
#&gt; 
#&gt; Slot "entrez_ids":
#&gt; </span><span style='color: #949494;'># A tibble: 3 x 3</span><span>
#&gt;   variant_id gene_name    entrez_id
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>      </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>        </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>    
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> rs4725504  DPP6         1804     
#&gt; </span><span style='color: #BCBCBC;'>2</span><span> rs4725504  PAXBP1P1     106481704
#&gt; </span><span style='color: #BCBCBC;'>3</span><span> rs4725504  LOC107986750 107986750
#&gt; </div><div class='input'>
<span class='co'># Or by its position in table variants</span>
<span class='no'>variants_ex01</span>[<span class='fl'>3</span>]</div><div class='output co'>#&gt; An object of class "variants"
#&gt; Slot "variants":
#&gt; </span><span style='color: #949494;'># A tibble: 1 x 7</span><span>
#&gt;   variant_id merged functional_class chromosome_name chromosome_posi…
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>       </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>            </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>                      </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span>
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> rs4725504       0 intergenic_vari… 7                      153</span><span style='text-decoration: underline;'>608</span><span>338
#&gt; </span><span style='color: #949494;'># … with 2 more variables: chromosome_region </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span style='color: #949494;'>, last_update_date </span><span style='color: #949494;font-style: italic;'>&lt;dttm&gt;</span><span>
#&gt; 
#&gt; Slot "genomic_contexts":
#&gt; </span><span style='color: #949494;'># A tibble: 4 x 11</span><span>
#&gt;   variant_id gene_name chromosome_name chromosome_posi… distance is_closest_gene
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>      </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>     </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>                      </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span>    </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span>          
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> rs4725504  DPP6      7                      153</span><span style='text-decoration: underline;'>608</span><span>338   </span><span style='text-decoration: underline;'>278</span><span>759 TRUE           
#&gt; </span><span style='color: #BCBCBC;'>2</span><span> rs4725504  DPP6      7                      153</span><span style='text-decoration: underline;'>608</span><span>338   </span><span style='text-decoration: underline;'>139</span><span>795 TRUE           
#&gt; </span><span style='color: #BCBCBC;'>3</span><span> rs4725504  PAXBP1P1  7                      153</span><span style='text-decoration: underline;'>608</span><span>338   </span><span style='text-decoration: underline;'>158</span><span>463 TRUE           
#&gt; </span><span style='color: #BCBCBC;'>4</span><span> rs4725504  LOC10798… 7                      153</span><span style='text-decoration: underline;'>608</span><span>338    </span><span style='text-decoration: underline;'>84</span><span>107 TRUE           
#&gt; </span><span style='color: #949494;'># … with 5 more variables: is_intergenic </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span style='color: #949494;'>, is_upstream </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span style='color: #949494;'>,
#&gt; #   is_downstream </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span style='color: #949494;'>, source </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span style='color: #949494;'>, mapping_method </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>
#&gt; 
#&gt; Slot "ensembl_ids":
#&gt; </span><span style='color: #949494;'># A tibble: 2 x 3</span><span>
#&gt;   variant_id gene_name ensembl_id     
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>      </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>     </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>          
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> rs4725504  DPP6      ENSG00000130226
#&gt; </span><span style='color: #BCBCBC;'>2</span><span> rs4725504  PAXBP1P1  ENSG00000233489
#&gt; 
#&gt; Slot "entrez_ids":
#&gt; </span><span style='color: #949494;'># A tibble: 3 x 3</span><span>
#&gt;   variant_id gene_name    entrez_id
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>      </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>        </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>    
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> rs4725504  DPP6         1804     
#&gt; </span><span style='color: #BCBCBC;'>2</span><span> rs4725504  PAXBP1P1     106481704
#&gt; </span><span style='color: #BCBCBC;'>3</span><span> rs4725504  LOC107986750 107986750
#&gt; </div><div class='input'>
<span class='co'># Keep all variants except the third</span>
<span class='no'>variants_ex01</span>[-<span class='fl'>3</span>]</div><div class='output co'>#&gt; An object of class "variants"
#&gt; Slot "variants":
#&gt; </span><span style='color: #949494;'># A tibble: 2 x 7</span><span>
#&gt;   variant_id merged functional_class chromosome_name chromosome_posi…
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>       </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>            </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>                      </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span>
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> rs1469924…      0 regulatory_regi… 8                       36</span><span style='text-decoration: underline;'>984</span><span>538
#&gt; </span><span style='color: #BCBCBC;'>2</span><span> rs56261590      0 intron_variant   4                       73</span><span style='text-decoration: underline;'>189</span><span>781
#&gt; </span><span style='color: #949494;'># … with 2 more variables: chromosome_region </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span style='color: #949494;'>, last_update_date </span><span style='color: #949494;font-style: italic;'>&lt;dttm&gt;</span><span>
#&gt; 
#&gt; Slot "genomic_contexts":
#&gt; </span><span style='color: #949494;'># A tibble: 17 x 11</span><span>
#&gt;    variant_id gene_name chromosome_name chromosome_posi… distance
#&gt;    </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>      </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>     </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>                      </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span>    </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span>
#&gt; </span><span style='color: #BCBCBC;'> 1</span><span> rs1469924… LOC10537… 8                       36</span><span style='text-decoration: underline;'>984</span><span>538    </span><span style='text-decoration: underline;'>81</span><span>555
#&gt; </span><span style='color: #BCBCBC;'> 2</span><span> rs1469924… KCNU1     8                       36</span><span style='text-decoration: underline;'>984</span><span>538    </span><span style='text-decoration: underline;'>48</span><span>413
#&gt; </span><span style='color: #BCBCBC;'> 3</span><span> rs1469924… LOC10537… 8                       36</span><span style='text-decoration: underline;'>984</span><span>538     </span><span style='text-decoration: underline;'>2</span><span>533
#&gt; </span><span style='color: #BCBCBC;'> 4</span><span> rs1469924… TPT1P8    8                       36</span><span style='text-decoration: underline;'>984</span><span>538    </span><span style='text-decoration: underline;'>95</span><span>231
#&gt; </span><span style='color: #BCBCBC;'> 5</span><span> rs1469924… RPL26P25  8                       36</span><span style='text-decoration: underline;'>984</span><span>538     </span><span style='text-decoration: underline;'>3</span><span>392
#&gt; </span><span style='color: #BCBCBC;'> 6</span><span> rs1469924… KCNU1     8                       36</span><span style='text-decoration: underline;'>984</span><span>538    </span><span style='text-decoration: underline;'>48</span><span>413
#&gt; </span><span style='color: #BCBCBC;'> 7</span><span> rs1469924… AC090453… 8                       36</span><span style='text-decoration: underline;'>984</span><span>538     </span><span style='text-decoration: underline;'>3</span><span>439
#&gt; </span><span style='color: #BCBCBC;'> 8</span><span> rs1469924… TPT1P8    8                       36</span><span style='text-decoration: underline;'>984</span><span>538    </span><span style='text-decoration: underline;'>95</span><span>008
#&gt; </span><span style='color: #BCBCBC;'> 9</span><span> rs1469924… AC092818… 8                       36</span><span style='text-decoration: underline;'>984</span><span>538    </span><span style='text-decoration: underline;'>82</span><span>903
#&gt; </span><span style='color: #BCBCBC;'>10</span><span> rs1469924… LOC10537… 8                       36</span><span style='text-decoration: underline;'>984</span><span>538    </span><span style='text-decoration: underline;'>59</span><span>331
#&gt; </span><span style='color: #BCBCBC;'>11</span><span> rs56261590 ANKRD17   4                       73</span><span style='text-decoration: underline;'>189</span><span>781        0
#&gt; </span><span style='color: #BCBCBC;'>12</span><span> rs56261590 ANKRD17   4                       73</span><span style='text-decoration: underline;'>189</span><span>781        0
#&gt; </span><span style='color: #BCBCBC;'>13</span><span> rs56261590 HMGA1P2   4                       73</span><span style='text-decoration: underline;'>189</span><span>781    </span><span style='text-decoration: underline;'>90</span><span>636
#&gt; </span><span style='color: #BCBCBC;'>14</span><span> rs56261590 HMGA1P2   4                       73</span><span style='text-decoration: underline;'>189</span><span>781    </span><span style='text-decoration: underline;'>89</span><span>275
#&gt; </span><span style='color: #BCBCBC;'>15</span><span> rs56261590 LOC10272… 4                       73</span><span style='text-decoration: underline;'>189</span><span>781    </span><span style='text-decoration: underline;'>69</span><span>394
#&gt; </span><span style='color: #BCBCBC;'>16</span><span> rs56261590 AC053527… 4                       73</span><span style='text-decoration: underline;'>189</span><span>781    </span><span style='text-decoration: underline;'>69</span><span>428
#&gt; </span><span style='color: #BCBCBC;'>17</span><span> rs56261590 RF00334   4                       73</span><span style='text-decoration: underline;'>189</span><span>781    </span><span style='text-decoration: underline;'>74</span><span>179
#&gt; </span><span style='color: #949494;'># … with 6 more variables: is_closest_gene </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span style='color: #949494;'>, is_intergenic </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span style='color: #949494;'>,
#&gt; #   is_upstream </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span style='color: #949494;'>, is_downstream </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span style='color: #949494;'>, source </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span style='color: #949494;'>, mapping_method </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>
#&gt; 
#&gt; Slot "ensembl_ids":
#&gt; </span><span style='color: #949494;'># A tibble: 8 x 3</span><span>
#&gt;   variant_id  gene_name  ensembl_id     
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>       </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>      </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>          
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> rs146992477 KCNU1      ENSG00000215262
#&gt; </span><span style='color: #BCBCBC;'>2</span><span> rs146992477 TPT1P8     ENSG00000219491
#&gt; </span><span style='color: #BCBCBC;'>3</span><span> rs146992477 AC090453.1 ENSG00000243503
#&gt; </span><span style='color: #BCBCBC;'>4</span><span> rs146992477 AC092818.1 ENSG00000254038
#&gt; </span><span style='color: #BCBCBC;'>5</span><span> rs56261590  ANKRD17    ENSG00000132466
#&gt; </span><span style='color: #BCBCBC;'>6</span><span> rs56261590  HMGA1P2    ENSG00000248641
#&gt; </span><span style='color: #BCBCBC;'>7</span><span> rs56261590  AC053527.1 ENSG00000250220
#&gt; </span><span style='color: #BCBCBC;'>8</span><span> rs56261590  RF00334    ENSG00000221093
#&gt; 
#&gt; Slot "entrez_ids":
#&gt; </span><span style='color: #949494;'># A tibble: 9 x 3</span><span>
#&gt;   variant_id  gene_name    entrez_id
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>       </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>        </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>    
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> rs146992477 LOC105379377 105379377
#&gt; </span><span style='color: #BCBCBC;'>2</span><span> rs146992477 KCNU1        157855   
#&gt; </span><span style='color: #BCBCBC;'>3</span><span> rs146992477 LOC105379376 105379376
#&gt; </span><span style='color: #BCBCBC;'>4</span><span> rs146992477 TPT1P8       59347    
#&gt; </span><span style='color: #BCBCBC;'>5</span><span> rs146992477 RPL26P25     100270966
#&gt; </span><span style='color: #BCBCBC;'>6</span><span> rs146992477 LOC105379375 105379375
#&gt; </span><span style='color: #BCBCBC;'>7</span><span> rs56261590  ANKRD17      26057    
#&gt; </span><span style='color: #BCBCBC;'>8</span><span> rs56261590  HMGA1P2      171559   
#&gt; </span><span style='color: #BCBCBC;'>9</span><span> rs56261590  LOC102724832 102724832
#&gt; </div><div class='input'>
</div></span></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#value">Value</a></li>
      
      <li><a href="#examples">Examples</a></li>
    </ul>

  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Ramiro Magno, Ana-Teresa Maia, Centre for Biomedical Research, University of Algarve.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
   </div>

  

  </body>
</html>

